encoder/parquet.go

```diff
@@ -0,0 +1,289 @@
+package encoder
+
+import (
+	"bytes"
+	"encoding/json"
+	"errors"
+	"fmt"
+	goparquet "github.com/fraugster/parquet-go"
+	"github.com/fraugster/parquet-go/parquet"
+	"github.com/fraugster/parquet-go/parquetschema"
+	cdl "github.com/mimiro-io/common-datalayer"
+	"io"
+	"strings"
+	"time"
+)
+
+func NewParquetItemFactory() ItemFactory {
+	return &ParquetItemFactory{}
+}
+
+type ParquetItemFactory struct{}
+
+func (c *ParquetItemFactory) NewItem() cdl.Item {
+	return &ParquetItem{data: make(map[string]any)}
+}
+
+type ParquetEncoderConfig struct {
+	SchemaDef      *parquetschema.SchemaDefinition `json:"schema"`
+	IgnoreColumns  []string                        `json:"ignore_columns"`
+	FlushThreshold int64                           `json:"flush_threshold"`
+}
+
+func NewParquetEncoderConfig(sourceConfig map[string]any) (*ParquetEncoderConfig, error) {
+	if sourceConfig["schema"] == nil {
+		return nil, errors.New("no schema specified in source config")
+	}
+	var err error
+	sourceConfig["schema"], err = parquetschema.ParseSchemaDefinition(sourceConfig["schema"].(string))
+	if err != nil {
+		return nil, err
+	}
+	data, err := json.Marshal(sourceConfig)
+	if err != nil {
+		return nil, err
+	}
+	config := &ParquetEncoderConfig{}
+	err = json.Unmarshal(data, config)
+	if err != nil {
+		return nil, err
+	}
+
+	return config, nil
+}
+
+func NewParquetDecoderConfig(sourceConfig map[string]any) (*ParquetEncoderConfig, error) {
+	if sourceConfig["schema"] == nil {
+		return nil, errors.New("no schema specified in source config")
+	}
+	var err error
+	sourceConfig["schema"], err = parquetschema.ParseSchemaDefinition(sourceConfig["schema"].(string))
+	if err != nil {
+		return nil, err
+	}
+	var flushThreshold int64
+	if sourceConfig["flush_threshold"] == 0 {
```

Comment:
This can be done after doing the parsing. That would be cleaner as works with the struct and no need for accessing the map.